sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -1319,39 +1320,34 @@ def test_enet_cv_sample_weight_correctness(fit_intercept, sparse_container):
         X = sparse_container(X)
     params = dict(tol=1e-6)
 
-    # Set alphas, otherwise the two cv models might use different ones.
-    if fit_intercept:
-        alphas = np.linspace(0.001, 0.01, num=91)
-    else:
-        alphas = np.linspace(0.01, 0.1, num=91)
-
-    # We weight the first fold 2 times more.
-    sw[:n_samples] = 2
+    # We weight the first fold n times more.
+    sw[:n_samples] = rng.randint(0, 5, size=sw[:n_samples].shape[0])
     groups_sw = np.r_[
         np.full(n_samples, 0), np.full(n_samples, 1), np.full(n_samples, 2)
     ]
     splits_sw = list(LeaveOneGroupOut().split(X, groups=groups_sw))
-    reg_sw = ElasticNetCV(
-        alphas=alphas, cv=splits_sw, fit_intercept=fit_intercept, **params
-    )
+    reg_sw = ElasticNetCV(cv=splits_sw, fit_intercept=fit_intercept, **params)
     reg_sw.fit(X, y, sample_weight=sw)
 
     # We repeat the first fold 2 times and provide splits ourselves
     if sparse_container is not None:
         X = X.toarray()
-    X = np.r_[X[:n_samples], X]
+    X_rep = np.repeat(X, sw.astype(int), axis=0)
+    ##Need to know number of repitions made in total
```

Comment:
```suggestion
    # Inspect the total number of random repetitions so as to adjust the size of
    # the first cross-validation group accordingly.
```

Reply:
Actually, I think that computing the number of repetitions is not needed, see the other suggestions below.